browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -73,11 +74,36 @@ DECLARE_LAZY_MATCHER(tracker_appended_matcher,
 
 #undef DECLARE_LAZY_MATCHER
 
-void ApplyPotentialQueryStringFilter(const GURL& request_url,
-                                     std::string* new_url_spec) {
-  DCHECK(new_url_spec);
+void ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
   SCOPED_UMA_HISTOGRAM_TIMER("Brave.SiteHacks.QueryFilter");
-  std::string new_query = request_url.query();
+
+  if (!ctx->initiator_url.is_valid()) {
+    // Direct navigations (e.g. omnibar, bookmarks) are exempted.
+    return;
+  }
+
+  if (ctx->redirect_source.is_valid()) {
+    if (ctx->internal_redirect) {
```

Comment:
isn't `ctx->internal_redirect` always false if we have good `redirect_source`?

Reply:
No, because we generate `307`s as a result of the query filter for example. If we don't filter those out here, we end up looking at these requests twice.

Reply:
I mean that `redirect_source` is set to a non-empty value only if `ctx->internal_redirect` is false, and it seems that `ctx->internal_redirect` cannot change to `true` afterwards without changing `ctx->redirect_source`. This way, if `ctx->redirect_source` is non-empty, `ctx->internal_redirect` should be `false`
Am I wrong?

Reply:
I believe you're right, as long as we keep these in sync.